src/Data/CartItem/Cart.ts

```diff
@@ -52,9 +53,9 @@ export function numberOfCartItems(): number | null {
 export async function checkoutCart(): Promise<DataItem> {
   const cartItems: DataItem[] = await load(() => CartItem.all().take())
 
-  const products: DataItem[] = cartItems.map(
-    (item) => item.get('product') as DataItem,
-  )
+  const products: DataItem[] = cartItems
+    .map((item) => item.get('product'))
+    .filter(isDataItem)
```

Comment:
* [ ] How about `.filter((item) => item instanceof DataItem)`?

Reply:
`DataItem` is only defined as an abstract typescript class and is not available during the javascript runtime.

Reply:
What do you mean by "abstract typescript class"? Is there an issue or some other pointer that explains why typeof doesn't work for abstract classes?
The class itself is public API.

<img width="532" alt="image" src="https://github.com/user-attachments/assets/00934d75-7fa5-47e5-adc3-0b820158d4fd">

Reply:
The exact error is `TypeError: Right-hand side of 'instanceof' is not an object`.

*How to reproduce*
* Change `export function isDataItem(item: unknown): item is DataItem` to `return item instanceof DataItem` (and adjust the import if needed)
* Add something to the cart
* Try to "Request quote"

*Expected*
The checkout works regular

*Actual*
An error toast is shown:
<img width="1624" alt="Screenshot 2024-10-15 at 08 47 08" src="https://github.com/user-attachments/assets/7b7606a0-9b85-4d08-9ade-ed4808a898f5">

Stack-trace:

```
TypeError: Right-hand side of 'instanceof' is not an object
    at isDataItem (types.ts:29:10)
    at Array.filter (<anonymous>)
    at checkoutCart (Cart.ts:58:6)
    at async onClick (CheckoutButtonWidgetComponent.tsx:58:22)
```

Reply:
Hmm, I see. 

```ts
export type {
  DataClass,
  DataItem,
  DataScope,
} from 'scrivito_sdk/data_integration';
```

We're basically adding workaround for the SDK exporting only the type. Maybe we could change that?

Reply:
Striking the `type` from `export type` sounds like a good idea (originally introduced by me in #10342). I'll prepare a PR.